### PR TITLE
feat(element): Add getter and setter for input elements

### PR DESCRIPTION
**Pull request by uglow**
This PR tries to solve the inconsistency (referred to in the [FAQ](https://github.com/angular/protractor/blob/master/docs/faq.md#the-result-of-gettext-from-an-input-element-is-always-empty)) when querying and setting the value of input elements, by adding `getValue()` and `setValue()` methods to the `element` object. This makes working with text-based form fields much easier. 

Examples:
```js
var aboutbox = element(by.model('aboutbox'));

// Existing approach
aboutbox.clear().sendKeys('Long comment');
// This doesn't return the value: aboutbox.getText() , so use this instead:
expect(aboutbox.getAttribute('value')).toEqual('Long comment');

// NEW approach
aboutbox.setValue('Long comment');
expect(aboutbox.getValue()).toEqual('Long comment');

// Additionally, setValue() supports an optional boolean parameter, indicating
// whether to send a TAB-key to the control, causing the control to loose focus.
// This behaviour is really helpful when trying to test field validations that are
// triggered by change or blur events.

aboutbox.setValue('Long comment', false);  // default = no tab key sent
expect(aboutbox.getValue()).toEqual('Long comment');
expect(browser.driver.switchTo().activeElement().getAttribute('ng-model')).toEqual('aboutbox');

// Now send a TAB keypress after clearing & setting the value
aboutbox.setValue('Long comment', true);  // true = send tab key
expect(aboutbox.getValue()).toEqual('Long comment');
expect(browser.driver.switchTo().activeElement().getAttribute('ng-model')).not.toEqual('aboutbox');
```

[Review on Reviewable](https://reviewable.io/reviews/angular/protractor/1958)

### CODE REVIEW

**Comment by uglow** [comment]
googlebot, I love you, but you are giving me a lot of grief. I authored this commit.

**Comment by juliemr** [comment]
@uglow noted, no need to worry about it anymore :)

File: lib/element.js [hunk 1]
```diff
@@ -955,6 +955,49 @@ ElementFinder.prototype.allowAnimations = function(value) {
...
+/**
+ * Shortcut for setting the value of any element that you would normally
+ * use sendKeys() to set it's value (e.g. <input type="text|number|...">
+ *
+ * getValue() and setValue() provide API-symmetry when working with
+ * input elements.
+ *
+ * @alias element(locator).setValue('value') element(locator).clear().sendKeys('value');
+ *
+ * @param {...(string|!webdriver.promise.Promise<string>)} var_args The sequence
```
Comment (juliemr) [inline lib/element.js]: This comment seems off, this is just treated as a string.

**Comment by uglow** [comment]
@juliemr Update JSDoc to match code, and thanks for taming the googlebot :)

**Comment by elgalu** [comment]
Don't see how this PR solves the inconsistency mentioned:

> This is a webdriver quirk. input and textarea elements always have empty getText values

I would expect a patch to `getText()` so that if the element is a kind of input then returns `elm.getAttribute('value')` else fallback to regular `elm.getText()`

Regarding `setValue` I think the name is misleading since internally performs a `sendKeys` .. maybe `setText` would be better? Else first user thinking might be that is doing a client side `foundElm.setAttribute("value", "some value");`

**Comment by uglow** [comment]
@elgalu - you raise an interesting point, which I did consider when naming these functions. 

The reasons for choosing  get/set value instead of get/set text are:
- I wanted to have different methods for working with input & textarea elements, because they ARE different from the other kinds of elements. I see no virtue in pretending they are the same. 
- typically developers (who are the target audience of Protractor, rather then end-users) talk about the "value" of an input element, rather than the "text" of an input element. This is probably due to there being a "value" attribute on input elements.

**Comment by uglow** [comment]
Any updates?

**Comment by sjelin** [comment]
@juliemr We should talk about this PR before you go on sabbatical

**Comment by juliemr** [comment]
Thanks for all of the work on this PR. I think that right now, this is adding some small convenience, but not enough to justify the addition to the API surface in the core Protractor library. I'd love to see this available as an additional library!

**Pull request closed without merging**